src/utils/checkAuth.js

```diff
@@ -0,0 +1,34 @@
+import React from "react";
+import { connect } from "react-redux";
+import PropTypes from "prop-types";
+
+const checkAuth = Component => {
```

Comment:
Can you just create a parent component and pass it chirldren. https://medium.com/@markgituma/passing-data-to-props-children-in-react-5399baea0356

Reply:
Thanks, @abayo-luc for your suggestion, It looks like good ideas!
But can you tell me what is wrong with this approach?